oshmem/mca/spml/ucx/spml_ucx_component.c

```diff
@@ -492,27 +499,36 @@ static int mca_spml_ucx_component_fini(void)
         }
     }
 
-    /* delete all workers */
+    /* delete all workers and rkey_stores */
     for (i = 0; i < mca_spml_ucx.active_array.ctxs_count; i++) {
-        ucp_worker_destroy(mca_spml_ucx.active_array.ctxs[i]->ucp_worker[0]);
-        free(mca_spml_ucx.active_array.ctxs[i]->ucp_worker);
-        free(mca_spml_ucx.active_array.ctxs[i]);
+        ctx = mca_spml_ucx.active_array.ctxs[i];
+
+        mca_spml_ucx_rkey_store_destroy(&ctx->rkey_store);
+        ucp_worker_destroy(ctx->ucp_worker[0]);
+        free(ctx->ucp_worker);
+        free(ctx);
```

Comment:
maybe add common function for this? also use it for line 531-533,540

Reply:
fixed